src/Test.Utilities/CodeFixTestBase.cs

```diff
@@ -26,43 +24,72 @@ public abstract class CodeFixTestBase : DiagnosticAnalyzerTestBase
+
+        protected void VerifyBasicFix(string[] oldSources, string[] newSources, int? codeFixIndex = null, bool allowNewCompilerDiagnostics = false, bool onlyFixFirstFixableDiagnostic = false, TestValidationMode validationMode = DefaultTestValidationMode)
+        {
+            var runner = new CodeFixRunner(GetBasicDiagnosticAnalyzer(), GetBasicCodeFixProvider(), validationMode: validationMode);
+            VerifyFix(LanguageNames.VisualBasic, runner, oldSources, newSources, codeFixIndex, allowNewCompilerDiagnostics, onlyFixFirstFixableDiagnostic, allowUnsafeCode: false);
         }
 
         protected void VerifyBasicFixAll(string oldSource, string newSource, bool allowNewCompilerDiagnostics = false, TestValidationMode validationMode = DefaultTestValidationMode)
         {
-            VerifyFixAll(LanguageNames.VisualBasic, GetBasicDiagnosticAnalyzer(), GetBasicCodeFixProvider(), oldSource, newSource, allowNewCompilerDiagnostics, validationMode, false);
+            var runner = new CodeFixRunner(GetBasicDiagnosticAnalyzer(), GetBasicCodeFixProvider(), validationMode: validationMode);
+            VerifyFixAll(LanguageNames.VisualBasic, runner, new[] { oldSource }, new[] { newSource }, allowNewCompilerDiagnostics, allowUnsafeCode: false);
         }
 
-        private void VerifyFix(string language, DiagnosticAnalyzer analyzerOpt, CodeFixProvider codeFixProvider, string oldSource, string newSource, int? codeFixIndex, bool allowNewCompilerDiagnostics, bool onlyFixFirstFixableDiagnostic, TestValidationMode validationMode, bool allowUnsafeCode)
+        private void VerifyFix(string language, CodeFixRunner runner, string[] oldSources, string[] newSources, int? codeFixIndex, bool allowNewCompilerDiagnostics, bool onlyFixFirstFixableDiagnostic, bool allowUnsafeCode)
         {
-            Document document = CreateDocument(oldSource, language, allowUnsafeCode: allowUnsafeCode);
-            var newSourceFileName = document.Name;
+            Assert.True(oldSources.Length == newSources.Length, "Length of expected and actual sources arrays must match.");
+            Document[] documents = CreateDocuments(oldSources, language, allowUnsafeCode: allowUnsafeCode);
 
-            VerifyFix(document, analyzerOpt, codeFixProvider, newSource, newSourceFileName, ImmutableArray<TestAdditionalDocument>.Empty, codeFixIndex, allowNewCompilerDiagnostics, onlyFixFirstFixableDiagnostic, validationMode);
+            var project = documents.First().Project;
+            Solution newSolution;
+            if (onlyFixFirstFixableDiagnostic || codeFixIndex.HasValue)
+            {
+                newSolution = runner.ApplySingleFix(project, ImmutableArray<TestAdditionalDocument>.Empty, codeFixIndex.HasValue ? codeFixIndex.Value : 0, fixableDiagnosticIndex: 0);
+            }
+            else
+            {
+                newSolution = runner.ApplyFixesOneByOne(project.Solution, ImmutableArray<TestAdditionalDocument>.Empty, allowNewCompilerDiagnostics);
```

Comment:
Once we enable fixall testing for the batch fixer, we should also invoke `VerifyFixAll` on this code path so we get FixAll testing by default for all code fixer tests.

Reply:
I think we can even remove verification OneByOne having FixAll properly working.

---
In reply to: [160532580](https://github.com/dotnet/roslyn-analyzers/pull/1514#discussion_r160532580) [](ancestors = 160532580)